tests/unit/test_graph.cpp

```diff
@@ -0,0 +1,725 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Apache License, Version 2.0,
+* modified with the Commons Clause restriction.
+*/
+
+#include "../googletest/include/gtest/gtest.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "../../src/graph/graph.h"
+#include "../../deps/GraphBLAS/Include/GraphBLAS.h"
+#include "../../src/graph/node_iterator.h"
+#include "../../src/util/simple_timer.h"
+#include "../../src/arithmetic/tuples_iter.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+// #include <time.h>
+// #include <stdio.h>
+// #include <string.h>
+// #include <stdlib.h>
+// #include "assert.h"
+
+// Console text colors for benchmark printing
+#define KGRN  "\x1B[32m"
+#define KRED  "\x1B[31m"
+#define KNRM  "\x1B[0m"
+
+// OK(method) is a macro that calls a GraphBLAS method and checks the status;
+// if a failure occurs, it prints the detailed error message.
+#define OK(method)                                          \
+{                                                           \
+    info = method ;                                         \
+    if (! (info == GrB_SUCCESS || info == GrB_NO_VALUE))    \
+    {                                                       \
+        printf ("file %s line %d\n", __FILE__, __LINE__) ;  \
+        printf ("%s\n", GrB_error ( )) ;                    \
+        assert (false);                                     \
+    }                                                       \
+}
+
+class GraphTest: public ::testing::Test {
+  protected:
+    // These functions are run for each TEST_F invocation
+    // (where they had previously been run once)
+    void SetUp() override {
```

Comment:
I don't think we need to run GrB_init,  srand and GrB_finalize for every TEST_F
but only once for this entire unit test file